cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/etl/transform/ScriptTransform.java

```diff
@@ -110,6 +115,18 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws Ille
   public void initialize(TransformContext context) {
     metrics = context.getMetrics();
     logger = LoggerFactory.getLogger(ScriptTransform.class.getName() + " - Stage:" + context.getStageId());
+
+    // for Nashorn (Java 8+) support -- get method to convert ScriptObjectMirror to List
+    try {
+      Class<?> somClass = Class.forName("jdk.nashorn.api.scripting.ScriptObjectMirror");
+      somValuesMethod = somClass.getMethod("values");
+    } catch (NoSuchMethodException e) {
+      throw new RuntimeException(
+        "Failed to method ScriptObjectMirror#values() for converting ScriptObjectMirror to List", e);
```

Comment:
We should also include some pointer for what the user should do. Check that their Java includes nashorn?